src/main/java/uk/gov/justice/hmpps/prison/service/SentenceEnvelopeService.kt

```diff
@@ -75,48 +65,38 @@ class SentenceEnvelopeService(
       offenderBooking.offender.birthDate,
       WordUtils.capitalizeFully(offenderBooking.offender.lastName),
       offenderBooking.location.id,
-      offenderBooking.alerts.stream().filter { obj: OffenderAlert -> obj.isActive }
-        .map { offenderAlert: OffenderAlert? -> OffenderAlertTransformer.transformForOffender(offenderAlert) }
-        .collect(Collectors.toList()),
+      offenderBooking.alerts.filter { it.isActive }
+        .map { OffenderAlertTransformer.transformForOffender(it) },
     )
-    val sentenceAdjustments =
-      offenderBooking.getSentenceAdjustments().stream().filter { obj: SentenceAdjustmentValues -> obj.isActive }
-        .toList()
-    val bookingAdjustments =
-      offenderBooking.getBookingAdjustments().stream().filter { obj: BookingAdjustment -> obj.isActive }
-        .toList()
-    val sentences = bookingService.getSentenceAndOffenceDetails(offenderBooking.bookingId).stream()
-      .filter { obj: OffenderSentenceAndOffences? -> Objects.nonNull(obj) }
-      .filter { sentence: OffenderSentenceAndOffences -> sentence.sentenceStatus == "A" }.toList()
-    val containsFine = sentences.stream().anyMatch { obj: OffenderSentenceAndOffences -> obj.isAFine }
-    val containsFixedTermRecall =
-      sentences.stream().anyMatch { obj: OffenderSentenceAndOffences -> obj.isFixedTermRecallType }
-    val offenderFinePaymentDtoList = getFinesIfRequired(containsFine, offenderBooking.bookingId)
-    val fixedTermRecallDetails = getFixedTermRecall(containsFixedTermRecall, offenderBooking.bookingId)
-    val sentenceCalcDates = bookingService.getBookingSentenceCalcDatesV1_1(offenderBooking.bookingId)
+    val sentenceAdjustments = offenderBooking.getSentenceAdjustments().filter { it.isActive }
+    val bookingAdjustments = offenderBooking.getBookingAdjustments().filter { it.isActive }
+    val bookingId = offenderBooking.bookingId
+
+    val sentences = offenderBooking.sentences.filter {
+      it.status == "A" &&
+        !it.calculationType.calculationType.contains("AGG") &&
+        it.calculationType.category != "LICENCE"
+    }.filterNotNull().map { it.sentenceAndOffenceDetail }.sortedBy { it.sentenceSequence }
+
+    val offenderFinePaymentDtoList = sentences.takeIf { sentences.any { it.isAFine } }?.let {
+      bookingService.getOffenderFinePayments(bookingId)
+    } ?: emptyList()
+
+    val fixedTermRecallDetails = sentences
+      .takeIf { sentences.any { it.isFixedTermRecallType } }?.let {
+        offenderFixedTermRecallRepository.findById(bookingId)
```

Comment:
remove call to fixed term service.  This was doing a verify booking access for each booking, which we don't want to do.  Could have created a new method but decided to just bring in inline anyway